### PR TITLE
made new todo page responsive and added ability to visualize selected…

### PR DESCRIPTION
**Pull request by adexfit**
made new todo page responsive and added ability to visualize selected todo category and priority button, edited home fragment for responsiveness

### CODE REVIEW

File: app/src/main/java/com/solid/todolistapp/features/task/NewTodoActivity.kt [hunk 1]
```diff
@@ -88,26 +93,111 @@ class NewTodoActivity : AppCompatActivity(), View.OnClickListener {
...
                 category = binding.familyButton.text.toString()
                 categoryColor = R.color.family_background
+                activeCategoryBtn = "family_button"
             }
             R.id.school_button -> {
                 category = binding.schoolButton.text.toString()
                 categoryColor = R.color.school
+                activeCategoryBtn = "school_button"
+            }
+        }
```
Comment (solidtm) [inline app/src/main/java/com/solid/todolistapp/features/task/NewTodoActivity.kt:124]: Hi @adexfit great Job you did here, I love how you thought about implementing the changes I mentioned. 
We still have a little more fixes to do: 

1. There would be no need to override onClick() function here any more, and the same thing can be achieved on the button onClickListener. Hence, remove this overidden onClick() function and create a function named setUpListeners and move this code s in the onClick(), but this time using view binding instead.

2. You are doing a lot of work on the UI thread inside this changeActiveBtnColor() function. 
So when we talked about functions in Kotlin, I thought you about extension functions. Create a general extension function on a View or Button that sets the textColor and backgroundColor, and use that for the different when conditions.

**Review by solidtm — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/solid/todolistapp/features/task/NewTodoActivity.kt (hunk 1 above)
Comment (solidtm) [reply app/src/main/java/com/solid/todolistapp/features/task/NewTodoActivity.kt:124]: If you have any further questions, please don't hesitate to drop  in this thread, so we fast track.

Comment (solidtm) [reply app/src/main/java/com/solid/todolistapp/features/task/NewTodoActivity.kt:124]: This is an example: 

fun View.visibleIf(show: Boolean?) {
    visibility = if (show == true) View.VISIBLE
    else View.GONE
}

fun View.setColor(check: Boolean?, @ColorRes checkedColor: Int, @ColorRes uncheckedColor: Int) {
    if (check == true) {
        setBackgroundColor(ContextCompat.getColor(context, checkedColor))
    } else setBackgroundColor(ContextCompat.getColor(context, uncheckedColor))
}

fun View.setTColor(check: Boolean?, @ColorRes checkedColor: Int, @ColorRes uncheckedColor: Int) {
    if (check == true) {
        (ContextCompat.getColor(context, checkedColor))
    } else setBackgroundColor(ContextCompat.getColor(context, uncheckedColor))
}

fun View.show() {
    visibility = View.VISIBLE
}

fun View.gone() {
    visibility = View.GONE
}

**Comment by adexfit** [comment]
I have made the changes sir

**Pull request merged**